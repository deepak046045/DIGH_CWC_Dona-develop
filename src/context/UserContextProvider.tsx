'use client';

import React, {
  Dispatch,
  ReactNode,
  SetStateAction,
  createContext,
  useMemo,
  useState,
} from 'react';
import {
  NavigationOption,
  ungatedNavigationOptions,
} from '@/models/NavigationOptions';
import { Appointment } from '@/models/Appointment';
import { AlertType } from '@/models/AlertType';
import mockAppointments from '@/mocks/mockAppointments';
import { CarePartner } from '@/models/CarePartner';
import { mockCarePartners } from '@/mocks/mockCarePartners';

type UserContextDataType = {
  isLoggedIn: boolean;
  setIsLoggedIn: Dispatch<SetStateAction<boolean>>;
  activeView: NavigationOption;
  setActiveView: Dispatch<NavigationOption>;
  alert: AlertType;
  setAlert: Dispatch<SetStateAction<AlertType>>;
  appointments: Appointment[];
  setAppointments: Dispatch<SetStateAction<Appointment[]>>;
  carePartners: CarePartner[];
  setCarePartners: Dispatch<SetStateAction<CarePartner[]>>;
};

export const UserContext = createContext<UserContextDataType>({
  isLoggedIn: false,
  setIsLoggedIn: () => {},
  activeView: ungatedNavigationOptions[0],
  setActiveView: () => {},
  alert: { isOpen: false, message: '', level: '' },
  setAlert: () => {},
  appointments: mockAppointments,
  setAppointments: () => {},
  carePartners: mockCarePartners,
  setCarePartners: () => {},
});

type Props = {
  children: ReactNode;
};

export function UserContextProvider({ children }: Props) {
  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
  const [activeView, setActiveView] = useState<NavigationOption>(
    ungatedNavigationOptions[0]
  );
  const [alert, setAlert] = useState<AlertType>({
    isOpen: false,
    message: '',
    level: '',
  });
  const [appointments, setAppointments] =
    useState<Appointment[]>(mockAppointments);
  const [carePartners, setCarePartners] =
    useState<CarePartner[]>(mockCarePartners);

  const values = useMemo(
    () => ({
      isLoggedIn,
      setIsLoggedIn,
      activeView,
      setActiveView,
      alert,
      setAlert,
      appointments,
      setAppointments,
      carePartners,
      setCarePartners,
    }),
    [isLoggedIn, activeView, alert, appointments, carePartners]
  );

  return <UserContext.Provider value={values}>{children}</UserContext.Provider>;
}
