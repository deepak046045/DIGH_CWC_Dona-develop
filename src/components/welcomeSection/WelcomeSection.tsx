'use client';

import React, { useContext } from 'react';
import { LdsButton, LdsImage } from '@elilillyco/ux-lds-react';
import { UserContext } from '@/context/UserContextProvider';
import layout from '@/scss/themes/layout.module.scss';
import typography from '@/scss/themes/typography.module.scss';
import buttons from '@/scss/themes/buttons.module.scss';
import { copyTexts } from '@/constants/copyTexts';
import styles from './WelcomeSection.module.scss';

export default function WelcomeSection() {
  const { isLoggedIn } = useContext(UserContext);
  return (
    <div className={`${styles.background}`}>
      <div className={`${styles.welcomeSection} ${layout.gutter}`}>
        <div className={styles.image_wrapper}>
          <LdsImage
            className={styles.image}
            src="/images/elderly-group-smiling.png"
            alt="ElderlyGroupSmiling"
          />
        </div>
        <div className={styles.welcomeSectionRightSide}>
          <div
            className={`${styles.welcomeSectionHeaderText} ${typography.themeTypographyH1}`}
          >
            {!isLoggedIn
              ? copyTexts.WelcomeSection.welcomeHeaderTextUngated
              : copyTexts.WelcomeSection.welcomeHeaderTextGated}
          </div>
          <div className={`${styles.welcomeSectionSubHeaderText}`}>
            {!isLoggedIn
              ? copyTexts.WelcomeSection.welcomeSubHeaderTextUngated
              : copyTexts.WelcomeSection.welcomeSubHeaderTextGated}
          </div>
          {!isLoggedIn && (
            <LdsButton
              data-testid="sign-up-button"
              classes={`${buttons.primary}`}
            >
              Sign up now
            </LdsButton>
          )}
        </div>
      </div>
    </div>
  );
}
