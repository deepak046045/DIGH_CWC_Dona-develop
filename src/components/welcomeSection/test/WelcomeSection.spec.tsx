import React from 'react';
import { UserContext } from '@/context/UserContextProvider';
import { render, screen } from '@testing-library/react';
import WelcomeSection from '../WelcomeSection';

describe('WelcomeSection', () => {
  it('should render page when logged in', () => {
    const mockValue = {
      isLoggedIn: true,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <WelcomeSection />
      </UserContext.Provider>
    );

    expect(screen.queryByTestId('sign-up-button')).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the page when not logged in', () => {
    const mockValue = {
      isLoggedIn: false,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <WelcomeSection />
      </UserContext.Provider>
    );

    expect(screen.queryByTestId('sign-up-button')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
