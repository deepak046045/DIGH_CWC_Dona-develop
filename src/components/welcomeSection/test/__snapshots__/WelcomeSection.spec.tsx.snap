// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSection should render page when logged in 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="welcomeSection gutter"
    >
      <div
        class="image_wrapper"
      >
        <img
          alt="ElderlyGroupSmiling"
          class="image"
          src="/images/elderly-group-smiling.png"
        />
      </div>
      <div
        class="welcomeSectionRightSide"
      >
        <div
          class="welcomeSectionHeaderText themeTypographyH1"
        >
          Hi Elizabeth, welcome to Kisunla support
        </div>
        <div
          class="welcomeSectionSubHeaderText"
        >
          Support resources for your treatment journey. Set up your calendar of appointments, reminders, and care partners.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomeSection should render the page when not logged in 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="welcomeSection gutter"
    >
      <div
        class="image_wrapper"
      >
        <img
          alt="ElderlyGroupSmiling"
          class="image"
          src="/images/elderly-group-smiling.png"
        />
      </div>
      <div
        class="welcomeSectionRightSide"
      >
        <div
          class="welcomeSectionHeaderText themeTypographyH1"
        >
          Welcome to Kisunla support.
        </div>
        <div
          class="welcomeSectionSubHeaderText"
        >
          Support resources for your treatment journey. Register to set up a
      calendar of appointments, set up reminders, and assign care
      partners.
        </div>
        <button
          class="lds-button primary"
          data-testid="sign-up-button"
          type="button"
        >
          Sign up now
        </button>
      </div>
    </div>
  </div>
</div>
`;
